processing/src/main/java/org/apache/druid/query/filter/RangeFilter.java

```diff
@@ -0,0 +1,1025 @@
+    final byte upperStrictByte = this.isUpperStrict() ? (byte) 1 : 0x0;
+
+    return new CacheKeyBuilder(DimFilterUtils.RANGE_CACHE_ID)
+        .appendByte(boundType)
+        .appendByte(DimFilterUtils.STRING_SEPARATOR)
+        .appendString(column)
+        .appendByte(DimFilterUtils.STRING_SEPARATOR)
+        .appendString(matchValueType.asTypeString())
+        .appendByte(DimFilterUtils.STRING_SEPARATOR)
+        .appendByteArray(upperBytes)
+        .appendByte(DimFilterUtils.STRING_SEPARATOR)
+        .appendByteArray(lowerBytes)
+        .appendByte(DimFilterUtils.STRING_SEPARATOR)
+        .appendByte(lowerStrictByte)
+        .appendByte(DimFilterUtils.STRING_SEPARATOR)
+        .appendByte(upperStrictByte)
+        .appendByte(DimFilterUtils.STRING_SEPARATOR)
+        .appendByteArray(extractionFn == null ? new byte[0] : extractionFn.getCacheKey())
+        .build();
+  }
+
+  @Override
+  public DimFilter optimize()
+  {
+    return this;
+  }
+
+  @Override
+  public Filter toFilter()
+  {
+    if (extractionFn != null) {
+      return new DimensionPredicateFilter(column, getPredicateFactory(), extractionFn, filterTuning);
+    }
+    return this;
+  }
+
+  @Override
+  public RangeSet<String> getDimensionRangeSet(String dimension)
+  {
+    // range partitioning converts stuff to strings.. so do that i guess
+
+    String lowerString = lowerEval.asString();
+    String upperString = upperEval.asString();
+    RangeSet<String> retSet = TreeRangeSet.create();
+    Range<String> range;
+    if (getLower() == null) {
+      range = isUpperStrict() ? Range.lessThan(upperString) : Range.atMost(upperString);
+    } else if (getUpper() == null) {
+      range = isLowerStrict() ? Range.greaterThan(lowerString) : Range.atLeast(lowerString);
+    } else {
+      range = Range.range(
+          lowerString,
+          isLowerStrict() ? BoundType.OPEN : BoundType.CLOSED,
+          upperString,
+          isUpperStrict() ? BoundType.OPEN : BoundType.CLOSED
+      );
+    }
+    retSet.add(range);
+    return retSet;
+  }
+
+  @Nullable
+  @Override
+  public BitmapColumnIndex getBitmapColumnIndex(ColumnIndexSelector selector)
+  {
+    if (!Filters.checkFilterTuningUseIndex(column, selector, filterTuning)) {
+      return null;
+    }
+    if (matchValueType.is(ValueType.STRING) && extractionFn == null) {
+      final ColumnIndexSupplier indexSupplier = selector.getIndexSupplier(column);
+      if (indexSupplier == null) {
+        return Filters.makeNullIndex(false, selector);
+      }
+      final LexicographicalRangeIndex rangeIndex = indexSupplier.as(LexicographicalRangeIndex.class);
+      if (rangeIndex != null) {
+        final String lower = hasLowerBound() ? lowerEval.asString() : null;
+        final String upper = hasUpperBound() ? upperEval.asString() : null;
+        if (NullHandling.isNullOrEquivalent(lower) && NullHandling.isNullOrEquivalent(upper)) {
+          return Filters.makeNullIndex(false, selector);
+        }
```

Comment:
in this case it is checking that both lower and upper bound are null, which maybe is just invalid, or i guess could match .. everything except null since there is no lower or upper bound?

Reply:
made null for both bounds just be a validation error